### PR TITLE
37 us 03.03.01 remove image

### LINKED ISSUE #37: US 03.03.01
As an administrator, I want to be able to remove images.

### LINKED ISSUE #110: TASK: When updating image ensure to delete the image first
- [x] In updating event poster
- [x] In updating profile picture
- [x] in update facility profile picture

### PR DESCRIPTION
**Pull request by SuperCreativeUniqueUsername**
US 03.03.01: As an administrator, I want to be able to remove images.
- **When images are deleted anywhere the reference must be set to "" and not null**

### Basic Stuff
- [x] UML
- [x] Storyboard (total refactor of admin, also want to thank Katharine for helping me with this)
- [x] Unit tests pass
- [x] Minimal UI testing

### Videos and Images
https://github.com/user-attachments/assets/f2cfb22e-1251-4890-91af-856f45204f63

image

### CODE REVIEW

File: app/src/androidTest/java/com/example/pygmyhippo/admin/AllEventsFragmentTest.java [hunk 1]
```diff
@@ -79,35 +79,36 @@ public void setup() {
...
+//            onView(withText(order)).check(ViewAssertions.matches(isDisplayed()));
+//        }
+//    }
+//
+//    @Test
+//    public void testAllEventList() {
+//        // TODO: Add more thorough testing for items in all list after navigation is figured out.
+//        onView(withId(R.id.a_alllist_recycler))
+//                .perform(RecyclerViewActions.actionOnItemAtPosition(0, click()));
+//    }
```
Comment (Kori-Kozicki) [inline app/src/androidTest/java/com/example/pygmyhippo/admin/AllEventsFragmentTest.java]: How much testing is left for this fragment? Do we test if stuff like the title displays at least?

File: app/src/main/java/com/example/pygmyhippo/admin/DeleteImageFragment.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+import com.example.pygmyhippo.database.ImageStorage;
+import com.example.pygmyhippo.database.StorageOnCompleteListener;
+import com.example.pygmyhippo.databinding.AdminDeleteImageBinding;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.squareup.picasso.Picasso;
+
+import java.util.ArrayList;
+
+public class DeleteImageFragment extends Fragment implements StorageOnCompleteListener<Uri> {
+
```
Comment (Kori-Kozicki) [inline app/src/main/java/com/example/pygmyhippo/admin/DeleteImageFragment.java:44]: Header comment and Javadocs 🥺???

File: app/src/main/java/com/example/pygmyhippo/admin/DeleteImageFragment.java [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+        backButton.setOnClickListener(view1 -> {
+            Log.d("EventFragment", "Back button pressed");
+            navController.popBackStack();
+        });
+
+        // set up the delete button
+        DeleteButton = view.findViewById(R.id.a_deleteImage_deleteBtn);
+        DeleteButton.setOnClickListener(view1 -> {
+            Log.d("EventFragment", "Back button pressed");
+
```
Comment (Kori-Kozicki) [inline app/src/main/java/com/example/pygmyhippo/admin/DeleteImageFragment.java:104]: Is this back  button logging supposed to be in the delete button?

**Review by Kori-Kozicki — APPROVED** [review]
Overall this looks really good! Just had some comments about comments lol

File: app/src/androidTest/java/com/example/pygmyhippo/admin/AllEventsFragmentTest.java (hunk 1 above)
Comment (SuperCreativeUniqueUsername) [reply app/src/androidTest/java/com/example/pygmyhippo/admin/AllEventsFragmentTest.java]: I have not idea, I honestly think it's good. there is no title on this fragment.

File: app/src/main/java/com/example/pygmyhippo/admin/DeleteImageFragment.java (hunk 2 above)
Comment (SuperCreativeUniqueUsername) [reply app/src/main/java/com/example/pygmyhippo/admin/DeleteImageFragment.java:44]: 🫡

File: app/src/main/java/com/example/pygmyhippo/admin/DeleteImageFragment.java (hunk 3 above)
Comment (SuperCreativeUniqueUsername) [reply app/src/main/java/com/example/pygmyhippo/admin/DeleteImageFragment.java:104]: oops

File: app/src/androidTest/java/com/example/pygmyhippo/admin/AllEventsFragmentTest.java (hunk 1 above)
Comment (SuperCreativeUniqueUsername) [reply app/src/androidTest/java/com/example/pygmyhippo/admin/AllEventsFragmentTest.java]: oh wait i miss read this, ill make one test lol

**Pull request merged**